daemon/cmd/daemon_main.go

```diff
@@ -1349,6 +1352,16 @@ func initEnv(vp *viper.Viper) {
 		}
 	}
 
+	if option.Config.EnableIPSec && option.Config.EnableIPSecEncryptedOverlay {
+		if !option.Config.TunnelingEnabled() {
+			log.WithError(err).Fatal("IPSec encrypted overlay requires tunneling to be enabled")
+		}
+	}
+
+	if option.Config.EnableIPSecEncryptedOverlay && !option.Config.EnableIPSec {
+		log.Warn("IPSec encrypted overlay is enabled but IPSec is not. Ignoring option.")
+	}
+
```

Comment:
We would also want to check that the TunnelProtocol is VXLAN, right?

Reply:
I didn't see the flag for this, actually, I think this particular setting is jammed inside a cell and not accessible from here, IIUC. Can you show me how I can sus out the TunnelProtocol? I did not see it anywhere in `option.Config`

Reply:
Fixed this in latest push. Winds up doing validation in `newDaemon` provides us access to the TunnelConfig object on the daemon's hive parameters.